### PR TITLE
minor: Add comments for `GroupedHashAggregateStream` struct

### PR DESCRIPTION
**Pull request by 2010YOUY01**
## Which issue does this PR close?

NA

## Rationale for this change

`GroupedHashAggregateStream` now has tens of fields due to the complexity of aggregation logic

This PR intends to categorize those fields into different groups and add comments for each group, to improve readability

## What changes are included in this PR?

reordering struct fields and adding comments

### CODE REVIEW

File: datafusion/physical-plan/src/aggregates/row_hash.rs [hunk 1]
```diff
@@ -76,35 +76,43 @@ use super::AggregateExec;
...
-    /// true when streaming merge is in progress
-    is_stream_merging: bool,
-
     /// aggregate_arguments for merging spilled data
     merging_aggregate_arguments: Vec<Vec<Arc<dyn PhysicalExpr>>>,
 
     /// GROUP BY expressions for merging spilled data
     merging_group_by: PhysicalGroupBy,
+
+    /* ======================================================================== */
```
Comment (comphead) [inline datafusion/physical-plan/src/aggregates/row_hash.rs:96]: I dont remember we use this kind of format. I would probably stick to a single format across the project. Or we fill the format can be improved lets do it everywhere and add the notes to the contributing guide

**Review by alamb — APPROVED** [review]
Thanks @2010YOUY01  -- I think this makes the code better

I think @comphead 's suggestions on making the commenting style uniform would be good to address as well

File: datafusion/physical-plan/src/aggregates/row_hash.rs (hunk 1 above)
Comment (2010YOUY01) [reply datafusion/physical-plan/src/aggregates/row_hash.rs:96]: I can't find datafusion style guide, so I'm assuming the practice is following the existing style
I searched the code base and found comment separator `// ========`, I'll update to keep consistent with this format

**Comment by alamb** [comment]
Thanks again @2010YOUY01  and @comphead

**Pull request merged**